pkg/provisioner/crdprovisioner.go

```diff
@@ -175,13 +181,18 @@ func NewCrdProvisioner(kubeConfig *rest.Config, objNamespace string) (*CrdProvis
 	if err != nil {
 		return nil, err
 	}
+	crdClient, err := crdClientset.NewForConfig(kubeConfig)
+	if err != nil {
+		return nil, err
+	}
 
 	kubeInformerFactory := informers.NewSharedInformerFactory(kubeClient, consts.DefaultInformerResync)
 	startWatchingKube(context.Background(), kubeInformerFactory)
 
 	azInformerFactory := azdiskinformers.NewSharedInformerFactory(azdiskClient, consts.DefaultInformerResync)
+	crdInformerFactory := crdInformers.NewSharedInformerFactory(crdClient, consts.DefaultInformerResync)
```

Comment:
Refactor the setup of all informers to a separate method `SetupInformers` that will initialize all the informers together and will wait for cache sync for all at the same time rather one at a time, i.e. ```cache.WaitForCacheSync(
		ctx.Done(),
                azInformerFactory.Informer().HasSynced,
                kubeInformerFactory.Informer().HasSynced, ...```
                
I am also not sure we currently wait for cacheSync for azInformerFactory.

Reply:
We wait for cacheSync for azInformerFactory [here](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/ed9c00b395b7e6c65ae0626dd0000a16586ba0f0/pkg/watcher/conditionwatcher.go#L70-L106).

Reply:
This is not related to your change, but there is an indirect dependency here. condition watcher starts the `azInformerFactory` that is initialized in this method, later CacheReader relies on the `azInformerFactory` to be started. So, maybe a better approach would be for watcher.New to initialize and return the started `azInformerFactory` that will be passed to `CachedReader`. Yet I think the best approach would be to initialize and start all controllers in one `SetupInformers` method and then pass them to appropriate components that rely on them.